xarray/core/variable.py

```diff
@@ -936,6 +936,48 @@ def shift(self, **shifts):
             result = result._shift_one_dim(dim, count)
         return result
 
+    def pad_with_fill_value(self, **pad_widths):
+        """
+        Return a new Variable with paddings.
+
+        Parameters
+        ----------
+        **pad_width: keyword arguments of the form {dim: (before, after)}
+            Number of values padded to the edges of each dimension.
+        """
+        if self.dtype.kind == 'b':
```

Comment:
Is there a better way to get an appropriate fill_value for non-float arrays?

Reply:
Maybe this function should take a fill value argument, which could default to `dtypes.NA`?